### DISCUSSION TITLE
Namespace记录重复

### DISCUSSION THREAD

**Discussion by jiangt22** [apolloconfig/apollo#3838] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/6244465/126750487-31dd91a2-66d0-4810-a886-687123abcf5a.png)
![image](https://user-images.githubusercontent.com/6244465/126751808-a4c77cc7-1e9f-448f-95b6-fc7194744051.png)
如上图，从数据上看入库时间是一致的（ApolloConfigDB中的重复记录入库时间是一致的），怀疑是并发所致，不知道大家是否有遇到类似的问题。目前记录都是逻辑删除，数据库层面也建不了唯一索引。

**Comment by Anilople** [comment, marked as answer]
从admin-service的源码来看

https://github.com/ctripcorp/apollo/blob/f9c69dbb181e9cb6c5cb0ee08ae915a52430c23a/apollo-adminservice/src/main/java/com/ctrip/framework/apollo/adminservice/controller/NamespaceController.java#L47-L56

并发场景下，确实有可能创建出2个同名的namespace，因为只是单机的如果不存在才添加

分布式场景或者并发场景下，是可能会出现这个问题的

**Reply by nobodyiam** [reply]
是的，目前存在这个隐患，#3700 有讨论
